### PR TITLE
Add `Add-PoshGitToProfile -AllUsers` support

### PR DESCRIPTION
**Pull request by dahlbyk**
Toward including an option to install posh-git with Git for Windows (https://github.com/git-for-windows/git/issues/1384), this adds an option to add to profile for all users.

Note that, as expected, `-AllUsers` from a non-elevated shell fails:

```
Add-Content : Access to the path
'C:\Windows\System32\WindowsPowerShell\v1.0\Microsoft.PowerShell_profile.ps1' is denied.
```
or
```
Add-Content : Access to the path
'C:\Windows\System32\WindowsPowerShell\v1.0\profile.ps1' is denied.
```

Running PowerShell as Administrator allows the global install.

### CODE REVIEW

**Comment by rkeithhill** [comment]
BTW we do have this code to determine if the process is elevated or not. 

https://github.com/dahlbyk/posh-git/blob/65658bf6992282460726d6f4a552bae92073b1db/src/GitPrompt.ps1#L115-L124

We could check if the process is not elevated and the user specified `-AllHosts` and give a better error message.  One that tells them they need to be running as admin in order to update AllHosts profiles.

**Comment by dahlbyk** [comment]
> We could check if the process is not elevated and the user specified -AllHosts and give a better error message.

Done.

> Installing posh-git for all users requires Administrator access.

File: src/Utils.ps1 [hunk 1]
```diff
@@ -104,9 +125,18 @@ function Add-PoshGitToProfile {
         $TestParams
     )
 
+    if ($AllUsers -and !(Test-Administrator)) {
+        throw 'Installing posh-git for all users requires an elevated host.'
```
Comment (rkeithhill) [inline src/Utils.ps1]: What do you think about rewording that as `'Adding posh-git to an AllUsers profile requires an elevated host.'`?

Comment (rkeithhill) [reply src/Utils.ps1]: Other than that, this PR looks good to me.

**Pull request merged**